src/crdt-clock.ts

```diff
@@ -1,18 +1,18 @@
 import { clockChangesSince } from "./crdt-helpers";
 import type { EncryptedBlockstore, CarTransaction } from "./storage-engine/";
-import type { DocUpdate, ClockHead } from "./types";
+import type { DocUpdate, ClockHead, IndexKeyType } from "./types";
 import { advance } from "@web3-storage/pail/clock";
 import { root } from "@web3-storage/pail/crdt";
 import { applyHeadQueue, ApplyHeadQueue } from "./apply-head-queue";
 
-export class CRDTClock {
+export class CRDTClock<T, K extends IndexKeyType> {
```

Comment:
I think this CRDT stuff should be decoupled from the DbIndex stuff that uses the index keys. Maybe there is another type here? Or maybe I'm missing something. This type might not be one that needs extensibility.